sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -1002,45 +1017,56 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
     return PI_INVALID_VALUE;
   }
 
-  cm_support::CmBuffer *CmBuf = nullptr;
-  cm_support::SurfaceIndex *CmIndex;
-
-  int Status = Context->Device->CmDevicePtr->CreateBuffer(
-      static_cast<unsigned int>(Size), CmBuf);
-
-  if (Status != cm_support::CM_SUCCESS) {
-    return PI_OUT_OF_HOST_MEMORY;
+  // Flag & HostPtr argument sanity check
+  if (!Context->checkSurfaceArgument(Flags, HostPtr)) {
+    return PI_INVALID_OPERATION;
   }
 
-  Status = CmBuf->GetIndex(CmIndex);
-  const std::lock_guard<sycl::detail::SpinLock> Lock{*PiESimdSurfaceMapLock};
-  assert(PiESimdSurfaceMap->find((unsigned int)CmIndex->get_data()) ==
-             PiESimdSurfaceMap->end() &&
-         "Failure from CM-managed buffer creation");
-
-  // Initialize the buffer with user data provided with 'HostPtr'
-  if ((Flags & PI_MEM_FLAGS_HOST_PTR_USE) != 0) {
-    if (HostPtr != nullptr) {
-      Status =
-          CmBuf->WriteSurface(reinterpret_cast<const unsigned char *>(HostPtr),
-                              nullptr, static_cast<unsigned int>(Size));
+  char *MapBasePtr = nullptr;
+  cm_buffer_ptr CmBuf;
+  cm_support::SurfaceIndex *CmIndex;
+  std::lock_guard<std::mutex> Lock(Context->CmSurfaceManageLock);
```

Comment:
Did not notice this before - please move this above line 1065, maybe in a separate PR